auth/server/src/main/java/com/seagates3/controller/AccountController.java

```diff
@@ -171,6 +190,22 @@ public ServerResponse create() {
           LOGGER.error("Create account delay failing - ", e);
           Thread.currentThread().interrupt();
         }
+        // Handle multi-threaded create API calls by deleting extra accounts,
+        // if account limit creation exceeded due to multiple thread/multiple
+        // node create() API calls.
+        try {
+          accountCount = accountDao.getTotalCountOfAccounts();
+
+          if (accountCount > maxAllowedLdapResults) {
+            // delete newly created account since we exceeded account
+            // creation limit.
+            accountDao.ldap_delete_account(account);
```

Comment:
can't we check accounts present already and not create only if limit is 1000 already

Reply:
that check is already there but we came across this scenario where 
lets say we have 999 accounts created in ldap and two parallel threads calls create() calls , in this case there are changes of two account creation .
hence to avoid that we added this extra check after account creation is done.

This scenario will occur because of concurrent request received by either multiple threads or multiple node/instance of authserver.